packages/strapi-utils/lib/content-types.js

```diff
@@ -84,10 +84,14 @@ const isCollectionType = ({ kind = COLLECTION_TYPE }) => kind === COLLECTION_TYP
 const isKind = kind => model => model.kind === kind;
 
 const getPrivateAttributes = (model = {}) => {
+  const hasCreatorFields = !model.uid.startsWith('strapi::') && model.modelType !== 'component';
```

Comment:
to avoid having to move the `hasCreateFields` & `createFieldsArePrivate` here, we could set `privateAttributes` as a getter:

```js
return {
  get privateAttributes() {
    return getPrivateAttributes()
  }
}
```